lightningd/plugin.c

```diff
@@ -1086,9 +1087,35 @@ void json_add_opt_plugins(struct json_stream *response,
 			  const struct plugins *plugins)
 {
 	struct plugin *p;
+	struct plugin_opt *opt;
+	const char *opt_name;
+	const char *plugin_name;
+	char *plugin_ext;
+
+	/* we output 'plugins' and their config as substructure:
+	 * "plugins" : { "name" -> { "param" : "value" }, ... }  */
+	json_object_start(response, "plugins");
 	list_for_each(&plugins->plugins, p, list) {
-		json_add_string(response, "plugin", p->cmd);
```

Comment:
if `deprecated_apis` is flagged on, we should still print this old style.

note that you'll want to add a `Changelog-Deprecated` note to the commit message for this.